include/ginkgo/core/matrix/coo.hpp

```diff
@@ -247,6 +247,11 @@ class Coo : public EnableLinOp<Coo<ValueType, IndexType>>,
     const LinOp* apply2(ptr_param<const LinOp> alpha, ptr_param<const LinOp> b,
                         ptr_param<LinOp> x) const;
 
+    /**
+     * Throws gko::Invalid exception if the data inside the arrays is invalid.
+     */
+    void validate_data() const;
```

Comment:
I think you can create a feature class contain validate_data virtual function and then override it here.